### PR TITLE
Feature/286 meta tags update

### PR DESCRIPTION
**Pull request by Niekvandam**
## Description

A few sentences describing the overall goals of the pull request's commits.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update

## Checklist

-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [X] I updated the changelog with an end-user readable description
-   [X] I assigned this pull request to the correct project board to update the sprint board

## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. Go to the homepage
2. Check if meta description in head is generated
3. Browse through the project overview / create / login / home pages
4. Check if title and meta description are relevant to the current page. 

## Link to issue

Closes: #286

### CODE REVIEW

File: src/app/modules/project/overview/overview.component.ts [hunk 1]
```diff
@@ -186,22 +189,27 @@ export class OverviewComponent implements OnInit {
...
 
     this.searchControl.valueChanges.subscribe((value) => this.onSearchInputValueChange(value));
 
     this.sortForm.valueChanges.subscribe((value) => this.onSortFormValueChange(value));
 
     this.highlightFormControl.valueChanges.subscribe((value) => this.onHighlightFormValueChanges(value));
 
+    // Updates meta and title tags
+    this.seoService.updateTitle('Project overview');
+    this.seoService.updateDescription('Browse or search for specific projects within DeX');
```
Comment (Brend-Smits) [inline src/app/modules/project/overview/overview.component.ts:210]:
```suggestion
    this.seoService.updateDescription('Browse or search for specific projects or ideas within DeX');
```

File: src/app/services/seo.service.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { stringify } from 'querystring';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SEOService {
+
+  constructor(private title: Title, private meta: Meta) { }
+
+  updateTitle(title: string) {
```
Comment (Brend-Smits) [inline src/app/services/seo.service.ts:12]: Missing access modifier (public, private) and return type.
Can you also provide comments to explain what the code does?

File: src/app/services/seo.service.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    this.title.setTitle(title + ' - DeX');
+  }
+
+  updateDescription(desc: string) {
```
Comment (Brend-Smits) [inline src/app/services/seo.service.ts:16]: Missing access modifier (public-private), as well as return types.
Can you also provide comments to explain what the code does?

**Review by Brend-Smits — CHANGES_REQUESTED** [review]
Looks great and works great.

File: src/app/services/seo.service.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+/*
+This service is used for SEO within the project (Search Engine Optimization). This means that
+all things related to modifying page metadata will be executed from this service.
+*/
+export class SEOService {
+
+  constructor(private title: Title, private meta: Meta) { }
+
+  // Updates the page title of the webpage
```
Comment (rubenb994) [inline src/app/services/seo.service.ts:17]: Please use JSDoc style documentation: https://jsdoc.app/about-getting-started.html
Example: 
```ts
/**
   I am JSDoc documentation
*/
public someMethod() { }
```

Comment (Niekvandam) [reply src/app/services/seo.service.ts]: I was looking on the website and was wondering: What comment style should I be using in a small code snippet (explaining a single line of code)? 
For example, should I be using method A or B. The difference in these being the way the comment is opened/closed
A:
```
// Check x for reasons y and z
if (x){
    //do stuff//
}
```

B:
```
/** Check x for reasons y and z */
if (x) {
    /** do stuff  */
 }
```

Comment (rubenb994) [reply src/app/services/seo.service.ts]: I would use option A for small code snippets. In general I use these as my guidelines:
- Method, class, interface, service -> Option B
- 1 or 2 lines of comment lines in the code -> Option A
- More than 2 lines of comments in the code -> Option B

**Review by Brend-Smits — APPROVED** [review]

File: src/app/modules/project/details/details.component.ts [hunk 5]
```diff
@@ -321,4 +326,6 @@ export class DetailsComponent implements OnInit {
     const timeZone = 'GMT';
     return dayOfTheWeek + ', ' + dateStamp + ', ' + timeStamp + ' ' + timeZone;
   }
+
+
```
Comment (Ruby77) [inline src/app/modules/project/details/details.component.ts:330]: Double enter?

File: src/app/modules/project/overview/overview.component.ts [hunk 6]
```diff
@@ -177,6 +179,7 @@ export class OverviewComponent implements OnInit {
     if (!environment.production) {
       this.displaySearchElements = true;
     }
+
```
Comment (Ruby77) [inline src/app/modules/project/overview/overview.component.ts:182]: Enter?

File: src/app/modules/project/overview/overview.component.ts [hunk 7]
```diff
@@ -186,22 +189,27 @@ export class OverviewComponent implements OnInit {
...
 
     this.sortForm.valueChanges.subscribe((value) => this.onSortFormValueChange(value));
 
     this.highlightFormControl.valueChanges.subscribe((value) => this.onHighlightFormValueChanges(value));
 
+    // Updates meta and title tags
+    this.seoService.updateTitle('Project overview');
+    this.seoService.updateDescription('Browse or search for specific projects or ideas within DeX');
+
+
```
Comment (Ruby77) [inline src/app/modules/project/overview/overview.component.ts:212]: Double enter?

File: src/app/modules/project/project.module.ts [hunk 8]
```diff
@@ -30,6 +30,8 @@ import { ModalHighlightDeleteComponent } from 'src/app/modules/project/modal-hig
 import { ModalHighlightComponent } from 'src/app/modules/project/modal-highlight/modal-highlight.component';
 import { SharedModule } from './../shared/shared.module';
 import { QuillModule } from 'ngx-quill';
+import { Meta } from '@angular/platform-browser';
+
```
Comment (Ruby77) [inline src/app/modules/project/project.module.ts:35]: Double enter

**Review by Ruby77 — APPROVED** [review]
Looks good😉

**Pull request merged**